1614 spring upgrade preparation replace javax classes with jakarta classes

camerojo opened the pull request:
WAIT FOR Mini Release 2.2.3 before merging to staging

Commit:
Add SpringBoot version 3.3.5 to gradle

Convert javax to jakarta

Commit:
Deleted org/tctalent/server/configuration/WebSocketSecurityConfig.java and replaced with EnableWebSocketSecurity on TcTalentApplication

Commit:
Replace javax.servlet with jakarta.servlet

Commit:
New DatabaseConfiguration replacing strategy

Commit:
Update to new Thymeleaf

Commit:
Replaced javax.transaction with jakarta

Commit:
Merge a standard SpringInitializr gradle build

camerojo committed:
Calculate isDev.

Remove unnecessary security dependencies given we have security starter

camerojo committed:
Calculate isDev and set as external property

Remove unnecessary security dependencies given we have security starter

camerojo committed:
Java8TimeDialect is now embedded into the core Thymeleaf - see https://www.thymeleaf.org/doc/articles/thymeleaf31whatsnew.html

camerojo committed:
jakartaActivation is a built in Boot dependency

camerojo committed:
Replace EmailSender with a Spring email version.

Commit:
Add in org.testcontainers

Commit:
New ES configuration

Commit:
Update doc of CandidateEsRepository

Commit:
Upgrade of Background task support to work with new Spring Trigger

Commit:
Deleted old EmailTester - no longer useful

Commit:
Change gradle build to allow Lombok to be used in testing code

Commit:
Delete old CandidateEsRepositoryTest.java - this will be replaced with something more useful.

Commit:
First cut at converting Elasticsearch code to use new Spring Data implementation

Lots of TODOs in here

sadatmalik commented on server/src/main/java/org/tctalent/server/service/db/impl/SavedSearchServiceImpl.java:
```diff
@@ -972,7 +962,7 @@ private Specification<Candidate> computeQuery(
     }
 
-    private BoolQueryBuilder computeElasticQuery(
+    private BoolQuery.Builder computeElasticQuery(
```
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 49. By comparison, 99% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test.

We recommend that you simplify this method or break it into multiple methods. For example, consider extracting the code block on lines 1134-1277 into a separate method.

camerojo committed:
Add missing security messaging dependency needed by websockets but not included in websocket starter

camerojo committed:
Add arg names to Auditor - was causing error in conversion to SpringBoot 3

Commit:
Can't use Flyway dependency generated by Spring Initializr

camerojo committed:
Add @Qualifier to remove ambiguity about which TemplateEngine to inject.

Spring Boot 3 does not allow this ambiguity.

camerojo committed:
No change - just reordered code

camerojo committed:
Add WebSocketSecurityConfig back in - even though it is a deprecated approach

camerojo committed:
No functional change. Doc only.

camerojo committed:
Doc only

camerojo committed:
Reinstate Postgres CandidateRepository wild card searches.

Reinstate CandidateService calling Postgres wild card searches.

Remove no longer needed ElasticsearchService equivalents (which no longer need to be ported across to new Elasticsearch API.

camerojo committed:
Now need to name the DestinationVariable

camerojo committed:
Remove @EnableWebSocketSecurity to match pre Spring Boot 3 code. Doesn't seem to be necessary here.

camerojo committed:
Compile with -parameters

Exclude commons-logging

camerojo committed:
Get logging of ElasticSearch query working

sadatmalik commented on server/src/main/java/org/tctalent/server/service/db/impl/SavedSearchServiceImpl.java:
```diff
@@ -972,7 +963,7 @@ private Specification<Candidate> computeQuery(
     }
 
-    private BoolQueryBuilder computeElasticQuery(
+    private BoolQuery.Builder computeElasticQuery(
```
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 49. By comparison, 99% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test.

We recommend that you simplify this method or break it into multiple methods. For example, consider extracting the code block on lines 1135-1278 into a separate method.

camerojo committed:
Refactor the actual ES search from SavedSearchServiceImpl to ElasticsearchServiceImpl

sadatmalik commented on server/src/main/java/org/tctalent/server/service/db/impl/SavedSearchServiceImpl.java:
```diff
@@ -972,7 +960,7 @@ private Specification<Candidate> computeQuery(
     }
 
-    private BoolQueryBuilder computeElasticQuery(
+    private BoolQuery.Builder computeElasticQuery(
```
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 49. By comparison, 99% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test.

We recommend that you simplify this method or break it into multiple methods. For example, consider extracting the code block on lines 1132-1275 into a separate method.

camerojo committed:
Move convert query to JSON call to ElasticsearchServiceImpl

sadatmalik commented on server/src/main/java/org/tctalent/server/service/db/impl/SavedSearchServiceImpl.java:
```diff
@@ -972,7 +959,7 @@ private Specification<Candidate> computeQuery(
     }
 
-    private BoolQueryBuilder computeElasticQuery(
+    private BoolQuery.Builder computeElasticQuery(
```
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 49. By comparison, 99% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test.

We recommend that you simplify this method or break it into multiple methods. For example, consider extracting the code block on lines 1131-1274 into a separate method.

camerojo committed:
Fix bug where null values were not handles in range filter.

Do some unit tests

Commit:
Refactored searchCandidateEs to take builder and PageRequest

sadatmalik commented on server/src/main/java/org/tctalent/server/service/db/impl/SavedSearchServiceImpl.java:
```diff
@@ -972,7 +958,7 @@ private Specification<Candidate> computeQuery(
     }
 
-    private BoolQueryBuilder computeElasticQuery(
+    private BoolQuery.Builder computeElasticQuery(
```
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 49. By comparison, 99% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test.

We recommend that you simplify this method or break it into multiple methods. For example, consider extracting the code block on lines 1130-1273 into a separate method.

Commit:
Doc only

camerojo committed:
Merge branch 'refs/heads/staging' into 1614-spring-upgrade-preparation-replace-javax-classes-with-jakarta-classes

camerojo committed:
First go at getting rid of using getExpressions in Specifications

Commit:
Improved model for use of Predicates

Commit:
Improved model for use of Predicates

camerojo committed:
Remove unnecessary cb.or

camerojo committed:
Converted remaining specifications to eliminate Predicate.getExpressions().add(... code

camerojo committed:
Don't add empty disjunctions to conjunctions (because they will always be false)

camerojo committed:
Replace deprecated rawStatusCode method

camerojo committed:
Set redundant Job opp fields

camerojo committed:
Enable Lombok for tests as well

camerojo committed:
Convert a couple of simple deprecations
